compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/AnnotationDesugar.java

```diff
@@ -383,21 +383,22 @@ private BLangLambdaFunction defineAnnotations(Location pos, BLangClassDefinition
         BLangRecordLiteral mapLiteral = null;
         boolean isLocalObjectCtor = classDef.flagSet.contains(Flag.OBJECT_CTOR);
         if (!classDef.annAttachments.isEmpty()) {
-            function = defineFunction(pos, pkgID, owner);
-            mapLiteral = ASTBuilderUtil.createEmptyRecordLiteral(pos, symTable.mapType);
-            addAnnotsToLiteral(classDef.annAttachments, mapLiteral, pos, env, isLocalObjectCtor);
+            Location annotPos = classDef.annAttachments.get(0).pos;
```

Comment:
Is this correct? Now we are passing the position of the first annotation attachment as the position of the lambda function we created. I don't think this is correct. 

As I remember, when generating the init function, we handled the position as a special case. Can't we do something similar here?

Reply:
It looks we create two lambda functions, one contains the positions of annotation attachments (which is actually needed for pinning the positions of annotations) and other one holds the whole position range of the construct which has the annotations. Init functions do not hold any actual file paths (empty string in that case). But this annotation desugared functions need to hold the actual file paths. 
So, in that case setting the position for the later lambda function as `new BLangDiagnosticLocation(pos.lineRange().fileName(), -1, -1, -1, -1, 0, 0)` also would work.

Reply:
As per the offline dicsusstion we are ok with getting first index since we do not have any other alternative.